src/System.Text.Json/src/System/Text/Json/Serialization/JsonCamelCaseNamePolicy.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.Json.Serialization
+{
+    internal sealed class JsonCamelCaseNamePolicy : JsonPropertyNamingPolicy
+    {
+        protected override string ConvertName(string name)
+        {
+            if (string.IsNullOrEmpty(name) || !char.IsUpper(name[0]))
+            {
+                return name;
+            }
+
+            char[] chars = name.ToCharArray();
```

Comment:
I don't know how much you care about perf here but I think you could use `String.Create` and avoid allocating a char[]. I haven't done it in Newtonsoft.Json because most older frameworks I target don't have it.